src/misc/utils.ts

```diff
@@ -24,17 +25,19 @@ export class Utils {
         Utils.isNode = typeof process !== 'undefined' && (process as any).release != null &&
             (process as any).release.name === 'node';
         Utils.isBrowser = typeof window !== 'undefined';
-        Utils.isNativeScript = !Utils.isNode && !Utils.isBrowser;
+        Utils.isWebWorker = !Utils.isBrowser && typeof global === 'undefined' && typeof self !== 'undefined';
+        Utils.isNativeScript = !Utils.isNode && !Utils.isBrowser && !Utils.isWebWorker;
         Utils.isMobileBrowser = Utils.isBrowser && this.isMobile(window);
         Utils.isAppleMobileBrowser = Utils.isBrowser && this.isAppleMobile(window);
-        Utils.global = Utils.isNativeScript ? global : (Utils.isNode && !Utils.isBrowser ? global : window);
+        Utils.global = Utils.isNativeScript ?
+            global : (Utils.isNode && !Utils.isBrowser && !Utils.isWebWorker ? global : self);
     }
 
     static fromB64ToArray(str: string): Uint8Array {
         if (Utils.isNode || Utils.isNativeScript) {
             return new Uint8Array(Buffer.from(str, 'base64'));
         } else {
-            const binaryString = window.atob(str);
+            const binaryString = self.atob(str);
```

Comment:
What is going on here with self? Is there a local atob function somewhere?

Reply:
`self` refers to the global object. It's useful because workers have their own global object instead of `window` - the MDN docs explain it well [here](https://developer.mozilla.org/en-US/docs/Web/API/Window/self):

>The `Window.self` read-only property returns the window itself, as a WindowProxy. It can be used with dot notation on a window object (that is, `window.self`) or standalone (`self`). The advantage of the standalone notation is that a similar notation exists for non-window contexts, such as in Web Workers. By using `self`, you can refer to the global scope in a way that will work not only in a window context (`self` will resolve to `window.self`) but also in a worker context (`self` will then resolve to `WorkerGlobalScope.self`).

So this is just another way of accessing the `atob`/`btoa` functions, which is compatible on the main thread (via `window`) and in workers (via `WorkerGlobalScope`).